base/namedtuple.jl

```diff
@@ -129,6 +129,12 @@ function NamedTuple{names, T}(nt::NamedTuple) where {names, T <: Tuple}
     end
 end
 
+# Like NamedTuple{names, T} as a constructor, but omits the additional
+# `convert` call, when the types are known to match the fields
+@eval function _new_NamedTuple(T::Type{<:NamedTuple}, args::Tuple)
```

Comment:
```suggestion
@eval function _new_NamedTuple(T::Type{NamedTuple{S, T}} where {S, T}, args::Tuple)
```
(this lowering forces it to be concrete)